MekHQ/src/mekhq/gui/dialog/ImageChoiceDialog.java

```diff
@@ -244,132 +200,67 @@ private void initComponents() {
-            scrBackgrounds.setViewportView(tableBackgrounds);
-            panelBackgrounds.add(scrBackgrounds, gbc);
-            backgroundsModel.reset();
-            backgroundsModel.setCategory(IconPackage.FORCE_BACKGROUNDS);
-            Iterator<String> imageNamesBackgrounds = imageItems.getItemNames(IconPackage.FORCE_BACKGROUNDS);
-            while (imageNamesBackgrounds.hasNext()) {
-                backgroundsModel.addImage(imageNamesBackgrounds.next());
+            LayeredForceIcon[] layeredForceIcons = LayeredForceIcon.values();
+            layeredTables = new JTable[layeredForceIcons.length];
+
+            for (int i = 0; i < layeredForceIcons.length; i++) {
+                JScrollPane scrollPane = new JScrollPane();
+                layeredTables[i] = new JTable();
+                ImageTableModel tableModel = new ImageTableModel();
+                JPanel panel = new JPanel();
+
+                layeredTables[i].setModel(tableModel);
+                layeredTables[i].setName(layeredForceIcons[i].getTableName());
+                layeredTables[i].setSelectionMode(layeredForceIcons[i].getListSelectionModel());
+                layeredTables[i].setRowHeight(76);
+                layeredTables[i].getColumnModel().getColumn(0).setCellRenderer(tableModel.getRenderer());
+                layeredTables[i].addMouseListener(new ImageTableMouseAdapter());
+                scrollPane.setViewportView(layeredTables[i]);
+                panel.add(scrollPane, gbc);
+                tableModel.reset();
+                tableModel.setCategory(layeredForceIcons[i].getLayerPath());
+                if (layeredForceIcons[i] != LayeredForceIcon.FRAME) {
+                    tableModel.addImage(Force.ICON_NONE);
+                }
+                Iterator<String> imageIterator = (imageItems != null)
+                        ? imageItems.getItemNames(layeredForceIcons[i].getLayerPath())
+                        : Collections.emptyIterator();
+                while (imageIterator.hasNext()) {
+                    tableModel.addImage(imageIterator.next());
+                }
+                layerTabs.addTab(layeredForceIcons[i].toString(), panel);
+
+                // Initialize Initial Values, provided the Icon Map is not empty
+                if (!iconMap.isEmpty()) {
+                    if (iconMap.containsKey(layeredForceIcons[i].getLayerPath())) {
+                        if (layeredForceIcons[i].getListSelectionModel() == ListSelectionModel.SINGLE_SELECTION) {
+                            // Determine the current selected value
+                            String selected = iconMap.get(layeredForceIcons[i].getLayerPath()).get(0);
+                            for (int k = 0; k < tableModel.getRowCount(); k++) {
+                                if (tableModel.getValueAt(k, 0).equals(selected)) {
+                                    layeredTables[i].setRowSelectionInterval(k, k);
```

Comment:
A comment as to why this is `, k` would help. The iteration makes it seem like it would be `, 0` or `, 1`.

Reply:
This is just because of how the backend is done, unfortunately. It actually sets it as having selected from k to k, inclusively, although the single selection only actually cares about the second value... Not the easiest logic to read, and I've added a comment for both places this occurs.